### PR TITLE
Lint class member order

### PR DESCRIPTION
**Pull request by edwardfoyle**
Adding [member ordering](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/member-ordering.md) lint rule and fixing related lint errors

No functional changes

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -11,3 +11,4 @@ expected-cdk-out
 .changeset/pre.json
 concurrent_workspace_script_cache.json
 scripts/components/api-changes-validator/test-resources/working-directory
+test-projects
```
Comment (edwardfoyle) [inline .prettierignore:14]: Not strictly related to this PR, but prettier doesn't need to run here

File: packages/create-amplify/src/npm_package_manager_controller.ts [hunk 2]
```diff
@@ -16,7 +16,6 @@ export class NpmPackageManagerController implements PackageManagerController {
     private readonly projectRoot: string,
     private readonly execa = _execa
   ) {}
-  private readonly executableName = 'npm';
```
Comment (edwardfoyle) [inline packages/create-amplify/src/npm_package_manager_controller.ts:19]: This was unused

File: .eslintrc.cjs [hunk 3]
```diff
@@ -83,6 +83,7 @@ module.exports = {
     ],
     '@typescript-eslint/restrict-template-expressions': 'error',
     '@typescript-eslint/consistent-type-definitions': ['error', 'type'],
+    '@typescript-eslint/member-ordering': 'error',
```
Comment (sobolk) [inline .eslintrc.cjs:86]: very nice

**Review by sobolk — APPROVED** [review]

File: .eslintrc.cjs (hunk 3 above)
Comment (edwardfoyle) [reply .eslintrc.cjs:86]: Thanks for the suggestion :)

**Review by 0618 — APPROVED** [review]

**Pull request merged**